rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -235,3 +241,84 @@ def find_uniswap_v2_lp_price(
     return Price(share_value)
+
+
+def decode_basic_uniswap_info(
+        amount_sent: int,
+        amount_received: int,
+        decoded_events: list[HistoryBaseEntry],
+        counterparty: str,
+        notify_user: Callable[[HistoryBaseEntry, str], None],
+) -> None:
+    """
+    Check last three events and if they are related to the swap, label them as such.
+    We check three events because potential events are: spend, (optionally) approval, receive.
+    Earlier events are not related to the current swap.
+    """
+    spend_event, approval_event, receive_event = None, None, None
+    for event in reversed(decoded_events):
+        try:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+        except (UnknownAsset, WrongAssetType):
+            notify_user(event, counterparty)
+            return
+
+        if (
+            event.event_type == HistoryEventType.INFORMATIONAL and
+            event.event_subtype == HistoryEventSubType.APPROVE and
+            approval_event is None
+        ):
+            approval_event = event
+        elif (
+            event.balance.amount == asset_normalized_value(amount=amount_sent, asset=crypto_asset) and  # noqa: E501
+            event.event_type == HistoryEventType.SPEND and
+            # don't touch ETH since there may be multiple ETH transfers
+            # and they are better handled by the aggregator decoder.
+            event.asset != A_ETH and
+            spend_event is None
+        ):
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.SPEND
+            event.counterparty = counterparty
+            event.notes = f'Swap {event.balance.amount} {crypto_asset.symbol} in {counterparty}'  # noqa: E501
+            spend_event = event
+        elif (
+            event.balance.amount == asset_normalized_value(amount=amount_received, asset=crypto_asset) and  # noqa: E501
+            event.event_type == HistoryEventType.RECEIVE and
+            event.asset != A_ETH and
+            receive_event is None
+        ):
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.counterparty = counterparty
+            event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} as a result of a {counterparty} swap'  # noqa: E501
+            receive_event = event
+        elif (
+            event.counterparty in {CPT_UNISWAP_V2, CPT_UNISWAP_V3} and
+            event.event_type == HistoryEventType.TRADE
+        ):
+            # The structure of swaps is the following:
+            # 1.1 Optional approval event
+            # 1.2 Optinal spend event
+            # 1.3 Optional receive event
+            # 1.4 SWAP_SIGNATURE event
+            # 2.1 Optional approval event
+            # 2.2 Optinal spend event
+            # 2.3 Optional receive event
+            # 2.4 SWAP_SIGNATURE event
```

Comment:
Impotant to specify there can be more

```suggestion
            # 2.4 SWAP_SIGNATURE event
            # etc.
```